packages/amplify_authenticator/lib/src/widgets/component.dart

```diff
@@ -50,7 +50,7 @@ abstract class StatelessAuthenticatorComponent extends StatelessWidget {
 /// {@template amplify_authenticator.authenticator_component}
 /// Base class for all stateful Authenticator components.
 /// {@endtemplate}
-abstract class AuthenticatorComponent<T extends AuthenticatorComponent<T>>
+abstract class AuthenticatorComponent<T extends StatefulWidget>
```

Comment:
Are these changes still necessary? Not seeing where it affects things

Reply:
Yes they are needed.

These changes are needed to make the changes to `AuthenticatorFormState`. Without those the changes to `AuthenticatorFormState`, `widget` within `_SignUpFormState` is of type `AuthenticatorForm` instead of the more specific `SignUpForm`.